File: app/src/main/java/com/tapan/weather/forecast/core/network/request/WeatherRequest.java
package com.tapan.weather.forecast.core.network.request;

public class WeatherRequest {

  private double latitude;
  private double longitude;

  private WeatherRequest(Builder builder) {
    this.latitude = builder.latitude;
    this.longitude = builder.longitude;
  }

  public double getLatitude() {
    return latitude;
  }

  public double getLongitude() {
    return longitude;
  }

  public static class Builder {

    private double latitude;
    private double longitude;

    public Builder() {
    }

    public Builder setLatitude(double latitude) {
      this.latitude = latitude;
      return this;
    }

    public Builder setLongitude(double longitude) {
      this.longitude = longitude;
      return this;
    }

    public WeatherRequest build() {
      return new WeatherRequest(this);
    }
  }
}

File: app/src/main/java/com/tapan/weather/forecast/core/network/request/WeatherRequest.kt
package com.tapan.weather.forecast.core.network.request

class WeatherRequest internal constructor(builder: Builder) {

  val latitude: Double
  val longitude: Double

  init {
    latitude = builder.latitude
    longitude = builder.longitude
  }

  class Builder {
    var latitude = 0.0
    var longitude = 0.0

    fun setLatitude(latitude: Double): Builder {
      this.latitude = latitude
      return this
    }

    fun setLongitude(longitude: Double): Builder {
      this.longitude = longitude
      return this
    }

    fun build(): WeatherRequest {
      return WeatherRequest(this)
    }
  }
}